pycue/tests/wrappers/allocation_test.py

```diff
@@ -0,0 +1,119 @@
+            hosts=host_pb2.HostSeq(
+                hosts=[host_pb2.Host(name=TEST_HOST_NAME)]))
+        getStubMock.return_value = stubMock
+
+        alloc = opencue.wrappers.allocation.Allocation(
+            facility_pb2.Allocation(name=TEST_ALLOC_NAME))
+        hosts = alloc.getHosts()
+
+        stubMock.GetHosts.assert_called_with(
+            facility_pb2.AllocGetHostsRequest(allocation=alloc.data), timeout=mock.ANY)
+        self.assertEqual(len(hosts), 1)
+        self.assertEqual(hosts[0].name(), TEST_HOST_NAME)
+
+    def testGetSubscriptions(self, getStubMock):
+        stubMock = mock.Mock()
+        stubMock.GetSubscriptions.return_value = facility_pb2.AllocGetSubscriptionsResponse(
+            subscriptions=subscription_pb2.SubscriptionSeq(
+                subscriptions=[subscription_pb2.Subscription(name=TEST_SUBSCRIPTION_NAME)]))
+        getStubMock.return_value = stubMock
+
+        alloc = opencue.wrappers.allocation.Allocation(
+            facility_pb2.Allocation(name=TEST_ALLOC_NAME))
+        subscriptions = alloc.getSubscriptions()
+
+        stubMock.GetSubscriptions.assert_called_with(
+            facility_pb2.AllocGetSubscriptionsRequest(allocation=alloc.data), timeout=mock.ANY)
+        self.assertEqual(len(subscriptions), 1)
+        self.assertEqual(subscriptions[0].name(), TEST_SUBSCRIPTION_NAME)
+
+    def testReparentHosts(self, getStubMock):
+        stubMock = mock.Mock()
+        stubMock.ReparentHosts.return_value = facility_pb2.AllocReparentHostsResponse()
+        getStubMock.return_value = stubMock
+
+        alloc = opencue.wrappers.allocation.Allocation(
+            facility_pb2.Allocation(name=TEST_ALLOC_NAME))
+        hosts = [host_pb2.Host(name=TEST_HOST_NAME)]
+        alloc.reparentHosts(hosts)
+
+        stubMock.ReparentHosts.assert_called_with(
+            facility_pb2.AllocReparentHostsRequest(
+                allocation=alloc.data, hosts=host_pb2.HostSeq(hosts=hosts)), timeout=mock.ANY)
+
+    def testSetName(self, getStubMock):
+        stubMock = mock.Mock()
+        stubMock.SetName.return_value = facility_pb2.AllocSetNameResponse()
+        getStubMock.return_value = stubMock
+
+        alloc = opencue.wrappers.allocation.Allocation(facility_pb2.Allocation())
+        alloc.setName(TEST_ALLOC_NAME)
+
+        stubMock.SetName.assert_called_with(
+            facility_pb2.AllocSetNameRequest(allocation=alloc.data, name=TEST_ALLOC_NAME),
+            timeout=mock.ANY)
+
+    def setTag(self, getStubMock):
+        stubMock = mock.Mock()
+        stubMock.SetTag.return_value = facility_pb2.AllocSetTagResponse()
+        getStubMock.return_value = stubMock
+
+        alloc = opencue.wrappers.allocation.Allocation(facility_pb2.Allocation())
+        alloc.setTag(TEST_ALLOC_TAG)
+
+        stubMock.SetTag.assert_called_with(
+            facility_pb2.AllocSetTagRequest(allocation=alloc.data, tag=TEST_ALLOC_TAG),
+            timeout=mock.ANY)
```

Comment:
It's not strictly necessary in our setup but it's nice to have a `__main__` block in these files that runs `unittest.main`.

Reply:
Thanks for catching that!